cpp/mg_utility/mg_utils.hpp

```diff
@@ -0,0 +1,209 @@
+/// @file mg_utils.hpp
+///
+/// The file contains methods that connect mg procedures and the outside code
+/// Methods like mapping a graph into memory or assigning new mg results or
+/// their properties are implemented.
+#pragma once
+
+#include <functional>
+
+#include "mg_graph.hpp"
+#include "mg_procedure.h"
+
+namespace mg_graph {
+
+template <typename TSize>
+void CreateGraphNode(mg_graph::Graph<TSize> *graph, const mgp_vertex *vertex) {
+  // Get Memgraph internal ID property
+  auto id_val = mgp_vertex_get_id(vertex);
+  TSize memgraph_id = id_val.as_int;
+
+  graph->CreateNode(memgraph_id);
+}
+
+template <typename TSize>
+void CreateGraphEdge(mg_graph::Graph<TSize> *graph,
+                     const mgp_vertex *vertex_from,
+                     const mgp_vertex *vertex_to) {
+  // Get Memgraph internal ID property
+  TSize memgraph_id_from = mgp_vertex_get_id(vertex_from).as_int;
+  TSize memgraph_id_to = mgp_vertex_get_id(vertex_to).as_int;
+  graph->CreateEdge(memgraph_id_from, memgraph_id_to);
+}
+} // namespace mg_graph
+
+namespace mg_utility {
+
+/// Calls a function in it's destructor (on scope exit).
+///
+/// Example usage:
+///
+/// void long_function() {
+///   resource.enable();
+///   // long block of code, might throw an exception
+///   resource.disable(); // we want this to happen for sure, and function end
+/// }
+///
+/// Can be nicer and safer:
+///
+/// void long_function() {
+///   resource.enable();
+///   OnScopeExit on_exit([&resource] { resource.disable(); });
+///   // long block of code, might trow an exception
+/// }
+class OnScopeExit {
+public:
+  explicit OnScopeExit(const std::function<void()> &function)
+      : function_(function) {}
+  ~OnScopeExit() { function_(); }
+
+private:
+  std::function<void()> function_;
+};
+
+///@brief Method for mapping the Memgraph in-memory graph into user-based graph
+/// with iterative node indices. Graph object stores information about node and
+/// edge mappings, alongside with connection information
+///
+///@param graph Memgraph graph
+///@param result Memgraph result object
+///@param memory Memgraph storage object
+///@return mg_graph::Graph
+///
+template <typename TSize = uint64_t>
+mg_graph::Graph<TSize> *GetGraphView(const mgp_graph *memgraph_graph,
```

Comment:
Avoid raw pointers as much as possible.
You can return `std::unique_ptr<mg_graph::Graph<TSize>>` here and bellow initialize with 
```
auto graph = std::make_unique<mg_graph::Graph<TSize>>();
```